### ISSUE TITLE
`parseJsonLogic` Not toggle adding extra combinator

### ISSUE THREAD

**Issue by smplman** [react-querybuilder/react-querybuilder#704]
### React Query Builder version

7.3.0

### Description

Ive been debugging an issue over the past couple of days where if the `Not` toggle is triggered and you take the resulting jsonLogic output and feed it back into `parseJsonLogic` it wraps an extra `and` group around the original logic.

I have traced my issue back to this block of code https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseJsonLogic/parseJsonLogic.ts#L127-L148. I'm not sure if there is an issue with my configuration or if this is indeed a bug.

### Reproduction

Using RQB with a not toggled and group and a test var here is the correct `jsonLogic` output:
```
{
  "!": {
    "and": [ {"var": "test" } ]
  }
}
```

If I then save this logic and then run it through `parseJsonLogic` to populate RQB here is the output with the extra layer of nesting.
```
{
  "and": [
    {
      "!": {
        "and": [ { "var": "test" } ]
      }
    }
  ]
}
```

**Comment by smplman** [comment]
Speaking with @jakeboone02 on discord he mentioned this could possibly be related to https://github.com/react-querybuilder/react-querybuilder/issues/690.

I also tried hacking around this issue by using custom `jsonLogicOperations` passed into `pareseJsonLogic`, but it seems like `!` and `and` are not being called.

Here is an examaple
```
const logic = {
  and: [
    { '==': [{ var: 'test' }, 4] }
  ],
  '!': { '==': [{ var: 'test' }, 4] },
  some: [
    { '==': [{ var: 'test' }, 4] }
  ],
  filter: [
    { '==': [{ var: 'test' }, 4] }
  ]
};

parseJsonLogic(logic, {
  jsonLogicOperations: {
    // not working
    and: value => console.log('and', value),
    '!': value => console.log('!', value),
    // working
    some: value => console.log('some', value),
    filter: value => console.log('filter', value)
  }
});
```

**Comment by smplman** [comment]
@jakeboone02 I took a look at this today and was able to change the `&&` to `||` on this line https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseJsonLogic/parseJsonLogic.ts#L143 and was able to resolve my issue.
```
} else if (isJsonLogicBetweenExclusive(logic['!']) || isRuleGroupType(rule)) {
```

I'm not positive if this is the best solution here, but all of the other tests continue to pass. Let me know if you would like for me to PR this change. I'm also not sure which `describe` my test below should belong to. Currently it's in `'valueSource: "value"'`

Here is the test I wrote to verify
```
  it('handles negated groups', () => {
    expect(parseJsonLogic({ '!': { 'and': [{ '==': [{ var: 'f1' }, 'Test'] }] } })).toEqual({
      combinator: 'and',
      not: true,
      rules: [{ field: 'f1', operator: '=', value: 'Test' }],
    });
  });
```
